src/NuGet.Protocol.Core.v3/DataClient/DataClient.cs

```diff
@@ -27,6 +27,14 @@ public sealed class DataClient : HttpClient
         private readonly INuGetMessageHandlerProvider[] _modifiers;
```

Comment:
Based on the fact that HttpHandlerResourceV3 has a HttpMessageHandler for retry and a HttpClientHandler for credentials, IMO, I don't see the need for INuGetMessageHandlerProvider. Any need for extensibility should likely be accomplished using HttpMessageHandler(s) that can be configured